src/batcher/batcher.rs

```diff
@@ -0,0 +1,85 @@
+use crate::batcher::solve_pair::solve_pair;
+use crate::batcher::submit_solution::submit_solution;
+use crate::models::orderbook::OrderBook;
+use anyhow::Result;
+use ethcontract::web3::types::Address;
+use std::collections::HashMap;
+
+pub async fn batch_process(orderbook: OrderBook) -> Result<()> {
+    let tokens: Vec<Address> = orderbook
+        .orders
+        .read()
+        .keys()
+        .into_iter()
+        .map(|t| Address::from(t.as_fixed_bytes()))
+        .collect();
+
+    let token_pairs = get_token_pairs(&tokens);
+    for token_pair in token_pairs {
+        let best_match = solve_pair(
+            orderbook
+                .orders
+                .read()
+                .get(&token_pair.0)
+                .unwrap_or(&HashMap::new())
+                .get(&token_pair.1)
+                .unwrap_or(&Vec::new()),
+            orderbook
+                .orders
+                .read()
+                .get(&token_pair.1)
+                .unwrap_or(&HashMap::new())
+                .get(&token_pair.0)
+                .unwrap_or(&Vec::new()),
+        )?;
+        submit_solution(best_match)?;
+    }
+    Ok(())
+}
+
+fn get_token_pairs(tokens: &Vec<Address>) -> Vec<(Address, Address)> {
```

Comment:
A slightly more idiomatic way would be to use iterators for this. Something like:
```rust
let len = tokens.len()
(0..len).flat_map(|i| (i..len).map(move |j| (tokens[i], tokens[j])))
```

This way, you can return an iterator (instead of collecting into a `Vec`):
```suggestion
fn get_token_pairs(tokens: &Vec<Address>) -> impl Iterator<Item = (Address, Address)> + '_ {
```

Also, in general, passing `Vec` references as function parameters [is considered unidomatic](https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg):

```suggestion
fn get_token_pairs(tokens: &[Address]) -> impl Iterator<Item = (Address, Address)> + '_ {
```